hearth/lib/hearth.rb

```diff
@@ -7,6 +7,7 @@
 require_relative 'hearth/config/env_provider'
 require_relative 'hearth/config/resolver'
 require_relative 'hearth/context'
+require_relative 'hearth/dns'
```

Comment:
RE: moving the patch out to here - I do think its somewhat net:http specific - I'd lean towards keeping it in our net:http client. (It can stay in the hearth/dns file, but maybe don't require it from here, require it from the client?).  Feel free to ignore - just a suggestion.

Reply:
Yeah. It's a good call out. If we add a new HTTP client, using Sockets, or even Typhoeus, curb, etc, we'd probably want this patch for this too. I think separate might make sense..